### ISSUE TITLE
Fix Member Of / Containing Fields

### ISSUE THREAD

**Issue by clarkepeterf** [project-lux/lux-marklogic#135]
**Problem Description:**

- There is no way to search for a set that is member of another set
- There is no way to search for a set that contains another set
- `containing` is commonly used but not marked as being the inverse of `memberOf`

**Expected Behavior/Solution:**
**_These fields exists right now_**:
- item `memberOf` field (E.g. item is member of a set)
- set `containing` field (e.g. a set contains an item)
- set `member` field (e.g. a set's member is an item)
  - this is a duplicate of `containing`

**_Set up the following fields_**:
- item `memberOf` (exisiting field) (E.g. item is member of a set)
- set `memberOf` (new field) (e.g. set is member of another set)
- set `containingItem` -inverse of the item `memberOf` field - (rename of existing 'containing' field) (e.g. set contains an item)
- set `containingSet` -inverse of the set `memberOf` field (new field) (e.g. set contains another set)

**Requirements:**
List of details required for the completion of the issue or requirements for the feature/bug. This can also include requirements that lie outside of the teams such as new design docs or clarification from an outside source.

**Needed for promotion:**
If an item on the list is not needed, it should be crossed off but not removed.

- ~[ ] Wireframe/Mockup - Mike~
- ~[ ] Committee discussions - Sarah~
- ~[ ] Feasibility/Team discussion - Sarah~
- ~[ ] Backend requirements - TBD~
- ~[ ] Frontend requirements- TBD~
- ~[ ] Questions~
- List of questions for discussions. Answers should be documented within the issue.

**UAT/LUX Examples:**

Sets/Collections now have both a "Containing Collections" and "Containing Objects" field in advanced search. Their "Member of" field is now fixed, which actually used to do the same thing as "Containing", which was incorrect

synopsis: 
- Member of goes from Object -> Set
- Containing Item Goes from Set -> Object
- Containing Set Goes from Set -> Other Set
- Member of goes from Set -> Set

Objects that are Member of: https://lux-front-tst.collections.yale.edu/view/results/objects?q=%7B%22memberOf%22%3A%7B%22name%22%3A%22art+in+focus+blue%22%7D%7D
 
Events Containing Objects:
https://lux-front-tst.collections.yale.edu/view/results/events?q=%7B%22used%22%3A%7B%22containingItem%22%3A%7B%22name%22%3A%22young+naval+officer%22%7D%7D%7D
 
Objects Containing Collections/Sets: https://lux-front-tst.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22memberOf%22%3A%7B%22AND%22%3A%5B%7B%22containingSet%22%3A%7B%22name%22%3A%22Boca+Raton%22%7D%7D%2C%7B%22classification%22%3A%7B%22name%22%3A%22Archive%22%7D%7D%5D%7D%7D%5D%7D

Object is the member of a set that contains a object: https://lux-front-tst.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22text%22%3A%22dog%22%2C%22_lang%22%3A%22en%22%7D%2C%7B%22memberOf%22%3A%7B%22containingItem%22%3A%7B%22name%22%3A%22badger%22%7D%7D%7D%5D%7D&ip=1&if=%7B%22AND%22%3A%5B%7B%22memberOf%22%3A%7B%22curatedBy%22%3A%7B%22OR%22%3A%5B%7B%22memberOf%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux-front-tst.collections.yale.edu%2Fdata%2Fgroup%2F6086b58d-941d-41e3-87c1-e00e96952ffb%22%7D%7D%2C%7B%22id%22%3A%22https%3A%2F%2Flux-front-tst.collections.yale.edu%2Fdata%2Fgroup%2F6086b58d-941d-41e3-87c1-e00e96952ffb%22%7D%5D%7D%7D%7D%2C%7B%22memberOf%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux-front-tst.collections.yale.edu%2Fdata%2Fset%2F5d41d8d9-3009-4e46-b8a5-6e376dd8537e%22%7D%7D%5D%7D&facetRequest=true

Object containing Set that contains a Set: https://lux-front-tst.collections.yale.edu/view/results/objects?q=%7B%22memberOf%22%3A%7B%22memberOf%22%3A%7B%22name%22%3A%22Florida%22%7D%7D%7D
 
Overview:
1. Go to https://lux-front-tst.collections.yale.edu/view/results/objects?q=
2. Choose "Have a single field" > "Member of" (this shows fields for Sets/Collections)
3. In the next "Have a single field" dropdown, there should be both a "Containing Collections" and "Containing Objects" dropdown. There used to only be a "Containing" dropdown for Objects, but Collections can also contain other Collections.
4. Alternatively, see the nested "Member of" which should now show show a Set/Collection list of fields rather than an Object set of fields

Supports https://github.com/project-lux/lux-middletier/issues/60 - if that middle tier item passes UAT, this item should as well
Also supports  https://github.com/project-lux/lux-frontend/issues/28 and https://github.com/project-lux/lux-middletier/issues/14

**Dependencies/Blocks:**

- Blocking: https://github.com/project-lux/lux-middletier/issues/60
- https://github.com/project-lux/lux-frontend/issues/28
- https://github.com/project-lux/lux-middletier/issues/14

**Related Github Issues:**

- Issues that contain similar work but are not blocking or being blocked by the current issue.

**Related links:**

- These links can consist of resources, bugherds, etc.

**Wireframe/Mockup:**
Place wireframe/mockup for the proposed solution at end of ticket.

**Comment by jffcamp** [comment]
Not reviewable on its own. Reviewable from https://github.com/project-lux/lux-middletier/issues/60

**Comment by azaroth42** [comment]
Additional UAT link:
https://lux-front-tst.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22memberOf%22%3A%7B%22curatedBy%22%3A%7B%22memberOf%22%3A%7B%22name%22%3A%22art+gallery%22%7D%7D%7D%7D%5D%7D 

This was the query that we found the issue with -- find all objects that are members of any set that is curated by any group that is part of a group called "art gallery" ... now finds 100k objects from YUAG

**Comment by roamye** [comment]
Approved per UAT

**Comment by roamye** [comment]
Looks good in prod, closing.
